background.js

```diff
@@ -46,83 +45,88 @@ chrome.runtime.onMessage.addListener(
 			default:
 				console.log(request);
 		}
-	});
-	
-if(initStorage("v0.8.92", true)) {
-	// v0.7
-	initStorage("popupcolor", "blue");
-	initStorage("highlight", true);
-	
-	// v0.8
-	// No changes to options
-	
-	// V0.8.5
-	initStorage("textboxhl", false);
+	})
+var getItems = ["popupcolor", "highlight", "textboxhl", "onlyreading",
+	"minihelp", "disablekeys", "kanjicomponents", "lineEnding", "copySeparator",
+	"maxClipCopyEntries", "popupDelay",
+	"showOnKey"];
+chrome.storage.sync.get(getItems,
+	function (items) {
+		if (initStorage("v0.8.92", true)) {
+			// v0.7
+			initStorage("popupcolor", "blue");
+			initStorage("highlight", true);
+
+			// v0.8
+			// No changes to options
 
-	// v0.8.6
-	initStorage("onlyreading", false);
-	// v0.8.8
-	if (localStorage['highlight'] == "yes")
-		localStorage['highlight'] = "true";
-	if (localStorage['highlight'] == "no")
-		localStorage['highlight'] = "false";
-	if (localStorage['textboxhl'] == "yes")
-		localStorage['textboxhl'] = "true";
-	if (localStorage['textboxhl'] == "no")
-		localStorage['textboxhl'] = "false";
-	if (localStorage['onlyreading'] == "yes")
-		localStorage['onlyreading'] = "true";
-	if (localStorage['onlyreading'] == "no")
-		localStorage['onlyreading'] = "false";
-	initStorage("copySeparator", "tab");
-	initStorage("maxClipCopyEntries", "7");
-	initStorage("lineEnding", "n");
-	initStorage("minihelp", "true");
-	initStorage("disablekeys", "false");
-	initStorage("kanjicomponents", "true");
+			// V0.8.5
+			initStorage("textboxhl", false);
 
-	for (i = 0; i*2 < rcxDict.prototype.numList.length; i++) {
-		initStorage(rcxDict.prototype.numList[i*2], "true")
-	}
+			// v0.8.6
+			initStorage("onlyreading", false);
+			// v0.8.8
+			if (items.highlight == "yes")

```

Comment:
These lines don't make sense. You're reading a value like "yes" from storage and then you're rewriting the same local object to have a different but similar value.

I know you were modeling it after what was happening to localStorage before but there are 2 main differences:
1) localStorage is a global variable so updating it's value actually changes real state instead of just local state.
2) localStorage was updated like that because I changed the names a long time ago and it was required for backwards compatibility. We don't need to do that for a brand new storage layer except when interacting with legacy storage.

So in this case, we don't need these new if statements but we may want the old ones for backwards compatibility, or we can argue that it's been long enough and drop support for the legacy yes no comments.

Reply:
Started this... 
A few more changes I have to make.

Reply:
For now I will revert back to localStorage

Reply:
Makes sense. I must of carelessly just changed it back when I didn't read all of the code.